Mathlib/Computability/CFG.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Computability.Language
+
+/-!
+# Context-Free Grammar
+
+This file contains the definition of a Context-Free Grammar (CFG), which is a grammar that has a
+single nonterminal symbol on the left-hand side of each rule.
+Derivation by a grammar is inherently nondeterministic.
+-/
+
+universe u
+
+/--
+The type of symbols is the disjoint union of terminals `T` and nonterminals `N`.
+-/
+inductive Symbol (T : Type u) (N : Type u)
+  | terminal    (t : T) : Symbol T N
+  | nonterminal (n : N) : Symbol T N
+
+/-- Context-free grammar that generates words over the alphabet `T` (a type of terminals). -/
+structure CFG (T : Type u) where
+  nt : Type u                            -- type of nonterminals
```

Comment:
Based on [this](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention), `NT` because it's an acronym? But ask Mario if you want to be sure.